containers/detail/types.ts

```diff
@@ -0,0 +1,11 @@
+import { withNavigation } from 'react-navigation';
+import { ExtractTInner } from '../../hocs/types';
+import { Todo } from '../../types/globalTypes';
+
+export interface Props {
+  todo: Todo[];
+  markAsNotDone: (todos: Todo[], index: number) => void;
```

Comment:
Can you think of a way of not defining this interface but extracting the `innerProps` of `mapDispatchToProps` & `mapStateToProps`?

Reply:
Can it be with the with the type ExtractTInner? I tried with that but I have a problems at runtime.

Reply:
sure, you need to type it as a `Hoc` though